test/modules/core/core/attribute.spec.js

```diff
@@ -120,3 +120,98 @@ test('WebGL#Attribute getValue', t => {
 
   t.end();
 });
+
+// If the vertex shader has more components than the array provides,
+// the extras are given values from the vector (0, 0, 0, 1) for the missing XYZW components.
+// https://www.khronos.org/opengl/wiki/Vertex_Specification#Vertex_format
+test('Attribute#missing component', t => {
+
+  if (fixture.gl.getExtension('STACKGL_resize_drawingbuffer')) {
+    // headless-gl does not seem to implement this behavior
+    t.comment('Skipping headless-gl');
+    t.end();
+    return;
+  }
+
+  const contexts = {
+    WebGL1: fixture.gl,
+    WebGL2: fixture.gl2
+  };
+
+  const modelProps = {
+    vs: `
+  attribute vec3 position;
```

Comment:
Let's see, should we test more variations of shader declarations?

`float`, `vec2`, ..., `mat3`, `mat4`?

Reply:
Looks like it only works for `vec3` and `vec4` (same in all browsers).